src/main.c

```diff
@@ -25,8 +27,8 @@ int main(int argc, char *argv[])
 	// Define energies	
 	energies = malloc(sizeof(double)*n_energies);
 	if(energies == NULL){
-		printf("Could not allocate energies memory.\n");
-		exit(1);
+		polycap_set_error(&error, POLYCAP_ERROR_MEMORY, "main: could not allocate memory for energies -> %s", strerror(errno));
+		return -1;
```

Comment:
Dit is nutteloos vrees ik, aangezien je dit doet in de `main()`: de error kan toch niet doorgegeven worden aan een andere functie. In dit geval is `return 1`, voorafgegaan door een foutmelding voldoende.

Reply:
AHzo. Alright. Ik ging ervanuit dat die set_error ook direct de fout weergaf, in welk geval het nog steeds nuttig zou zijn.

Vrijdag ben ik overigens effectief de hele dag thuis (enfin ja, toch niet op het werk). Kzal je dus helaas niet zien.

Reply:
Nope, `polycap_set_error` doet niet meer dan een struct alloceren en initializeren vrees ik.

Geen probleem voor vrijdag. Zal nog wel eens lukken. Is het trouwens weer bachelor project? Zou het kunnen dat ze iets doen met XMI-MSIM?

Reply:
Klopt helemaal. Weer kwantificatie van een Barringer meteoriet deeltje, mbv het monochromatisch labotoestel. Dus standaard opmeten, geometry goed zetten, meteoriet opmeten, kwantificeren. En dan die laatste stapjes herhalen tot ze klaar zijn ;)

Reply:
Lol... de hoeveelste keer is dat al dan? Wie doet de begeleiding?

Reply:
Nog maar tweede keer eigenlijk. Begeleiding door mij en Ella.
Kheb zelfs aan Laszlo voorgesteld om eigenlijk altijd dat als onderwerp te gebruiken. Na enkele jaren tijd hebben we dan een mooie dataset aan gekwantificeerde data omtrent die barringer meteoriet (want het zijn altijd nieuwe deeltjes), kunnen we dan wat leuke statistiek ofzo op uitvoeren. Kan mss nog interessant zijn.

Reply:
Nice... vinden de studenten het wat interessant? Laat maar weten als ze een bug zouden vinden in XMI-MSIM!

Reply:
Ja, de studenten hebben zeker interesse. Vaak krijgen we aantrok van die enkele chemisten die niet graag in het labo staan ;) Maar de meteorieten op zich trekken ook veel interesse. Om voorbeeld te geven, dit jaar waren er 15 projecten en slechts 14 studenten (7 groepjes) om ze te vullen, dus elke student krijgt basically waar ze voor kiezen, en toch werd ons project meermaals gekozen.

Op vlak van bugs vinden denk ik dat we eerder bugs in de studenten vinden, maar als we iets tegen komen laat ik het zeker weten ;)
Er is trouwens nog steeds geen automatische iteratieve modus in xmi-msim (of daarbuiten toevallig behalve in pymca) waar je een spectrum automatisch kan kwantificeren door XMI-MSIM simulaties zeker he (mits gekende geometrie)? Kvraag het omdat we momenteel gewoon nog altijd visueel vergelijken of het een 'goede overeenkomst' is, maar is wat arbitrair natuurlijk. Zou tof zijn om automatisch naar bvb een gemiddelde deviatie van 0.5% ofzo te rekenen.

Reply:
Cool, hopelijk komt er dan nog iets uit dat jullie later kunnen publiceren...

Wat betreft de quantification met AXIL: Ik herinner mij vaag dat @jangarrevoet daar ooit eens ging naar kijken maar weet niet of dat ook effectief gebeurd is. In elk geval is dat zeker te doen door u te baseren op `bin/xmimsim-pymca.c`. Misschien geen slecht idee voor een master thesis? Natuurlijk wordt het project voor de bachelor studenten dan wel een pak simpeler 😄

Reply:
Hij heeft daar naar gekeken, maar het werkte niet echt voor zover ik me herinner.
Kzal eens naar die `bin/xmimsim-pymca.c` kijken als ik wat tijd heb :)

Zou indeed het werk van de studenten wat wegnemen, maar het is niet omdat we het geautomatiseerd kunnen doen verlopen, dat we het hen niet eerst een paar keer manueel laten doen voor pedagogische doeleinden ;)